### PR TITLE
Integrate Postgres with the Play app using EBean.

### PR DESCRIPTION
**Pull request by hmhsu**
Connect the app to Postgres database in conf/application.conf.

Add EBean models under app/models/ and repository to use them under app/repository.
Add a PostgresController to allow a simple way to interact with Postgres from HTTP.
Add database evolution script for sample data. Also include tests.

Switch sbt version back to 1.3.13 due to issues with the new version (1.4.3).

### CODE REVIEW

File: universal-application-tool-0.0.1/app/models/PostgresDatabase.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package models;
+
+import javax.inject.*;
+
+import play.db.*;
```
Comment (bion) [inline universal-application-tool-0.0.1/app/models/PostgresDatabase.java:5]: let's follow the google java style guide and not use wildcard imports

**Comment by bion** [comment]
It's probably worth taking a look at our ORM docs for Postgres for this. We generally do not want to be using the `Database`, `PostgresDatabase`, or `DatabaseContext` classes directly.

**Comment by hmhsu** [comment]
@bion will do. sorry the PR is not ready for review yet. I will clean it up.

**Comment by hmhsu** [comment]
I think the PR is now review-ready.

File: universal-application-tool-0.0.1/test/FunctionalTest.java [hunk 2]
```diff
@@ -24,4 +46,42 @@ public void renderTemplate() {
     assertThat("text/html").isEqualTo(html.contentType());
     assertThat(html.body()).contains("Your new application is ready.");
   }
+
+  @Test
+  public void listPersons() {
+    Http.RequestBuilder request =
+        fakeRequest(routes.PostgresController.list())
+            .header(Http.HeaderNames.HOST, "localhost:" + play.api.test.Helpers.testServerPort());
```
Comment (ndmckinley) [inline universal-application-tool-0.0.1/test/FunctionalTest.java:54]: Can you import this function statically like the others?

File: universal-application-tool-0.0.1/conf/evolutions/default/2.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# --- Sample dataset
```
Comment (ndmckinley) [inline universal-application-tool-0.0.1/conf/evolutions/default/2.sql:1]: Fantastic!  Can you elaborate a little on how these are created and how to write more of these?  I assume it's the structured comments and their location under conf/evolutions/default that make them work, but if there's more to it it'd be great to have that written down.

**Comment by ndmckinley** [comment]
Wonderful, looks great to me, test looks good.

What happens if you try to launch it without the db?  Is that still workable, or do you now need the DB to be able to use any page?

**Comment by hmhsu** [comment]
> What happens if you try to launch it without the db? Is that still workable, or do you now need the DB to be able to use any page?

The server will start but once you connect to it, it will in turn try to connect to db and time out. So you won't be able to access any page without db. Not sure if that's okay or should we have a way to let app respond without a db connection.

File: universal-application-tool-0.0.1/conf/evolutions/default/2.sql (hunk 3 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/conf/evolutions/default/2.sql:1]: I followed the setup here https://github.com/playframework/play-samples/blob/2.8.x/play-java-ebean-example/conf/evolutions/default/2.sql.

Reading from the [documentation](https://www.playframework.com/documentation/2.8.x/Evolutions#Evolutions-scripts), the scripts for each database should locate in `conf/evolutions/{database name}` and numbered in name.

The relevant comments are `-- !Ups` and `-- !Downs`. The rest seems to be regular comments.

File: universal-application-tool-0.0.1/test/FunctionalTest.java (hunk 2 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/test/FunctionalTest.java]: Done.

File: universal-application-tool-0.0.1/conf/evolutions/default/2.sql (hunk 3 above)
Comment (ndmckinley) [reply universal-application-tool-0.0.1/conf/evolutions/default/2.sql:1]: Cool, works for me!  Can you add a section to `README.md` about this?  Some questions to answer:

- is the database created newly every time the app starts up?
- how do we add new schema elements / new tables / etc?
- how do we ensure that the db is up to date?
- if we need to undo a db schema change we're working on in dev, how do we do that?

**Review by ndmckinley — APPROVED** [review]

**Comment by natsid** [comment]
> > What happens if you try to launch it without the db? Is that still workable, or do you now need the DB to be able to use any page?
> 
> The server will start but once you connect to it, it will in turn try to connect to db and time out. So you won't be able to access any page without db. Not sure if that's okay or should we have a way to let app respond without a db connection.

Just to clarify: Will the DB be there now, after this merge? And do I need to do anything else when locally developing to connect to the DB?

**Comment by hmhsu** [comment]
> Just to clarify: Will the DB be there now, after this merge? And do I need to do anything else when locally developing to connect to the DB?

Yes, the DB will launch along side with app when bin/run-dev is run.

**Comment by natsid** [comment]
> > Just to clarify: Will the DB be there now, after this merge? And do I need to do anything else when locally developing to connect to the DB?
> 
> Yes, the DB will launch along side with app when bin/run-dev is run. If you wish to add schema to db, please see files under models/ and repository/ as example. Also you will likely need to update evolution scripts under conf/evolutions/default.

OK, thanks!

**Comment by bion** [comment]
Great work!

It'd be good to get an example of handling a request with synchronous code in (probably not in this PR). The code should be quite a bit simpler e.g. 

```
Person person = new Person();
person.setName(name);
person.save!
```

**Comment by hmhsu** [comment]
> It'd be good to get an example of handling a request with synchronous code in (not necessarily in this PR). The code should be quite a bit simpler e.g.

Good suggestion. I can take a look at that.

**Pull request merged**